src/bin/cargo/commands/add.rs

```diff
@@ -161,20 +155,29 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     let section = parse_section(args);
 
     let ws = args.workspace(config)?;
+
+    if args.is_present_with_zero_values("package") {
+        print_available_packages(&ws)?;
+    }
+
     let packages = args.packages_from_flags()?;
     let packages = packages.get_packages(&ws)?;
     let spec = match packages.len() {
         0 => {
             return Err(CliError::new(
-                anyhow::format_err!("no packages selected.  Please specify one with `-p <PKGID>`"),
+                anyhow::format_err!("no packages selected to add dependencies.  Please specify one with `-p <PKGID>`"),
```

Comment:
> "to add dependencies"

Isn't complete.  It'd at least be "to add dependencies to".  In general, I'd just simplify it to:

> "to modify"